edx_django_utils/monitoring/middleware.py

```diff
@@ -63,13 +63,17 @@ def accumulate_metric(cls, name, value):
         metrics_cache.set(name, value)
 
     @classmethod
-    def _batch_report(cls):
+    def _batch_report(cls, request):
         """
         Report the collected custom metrics to New Relic.
         """
         if not newrelic:
             return
         metrics_cache = cls._get_metrics_cache()
+        try:
+            newrelic.agent.add_custom_parameter('user_id', request.user.id)
```

Comment:
I see this is a rushed PR in trying to make a deadline.  So this comment is for a subsequent revision.

It seems odd to me that we are adding a _specific_ custom metric in a middleware that is providing _support_ for custom metrics - but not adding any specific ones itself.  We add specific custom metrics in other places in the code.  If we want to have a metric that we add to all of our APIs, for example, it should belong in [RequestMetricsMiddleware](https://github.com/edx/edx-drf-extensions/blob/0f69c806e85b84ad35efe0dd48cdf16619437972/edx_rest_framework_extensions/middleware.py#L7).  If my understanding of this separation of concerns is incorrect then we should update the docstring of this middleware.  Thanks.  :D

Reply:
@nasthagiri: What you say is true.  I must admit, I didn't really review the code - even though I merged.  I was counting on the other set of approvals.  Sorry.
@calistenson: Is this a refactor you can do?

Reply:
@nasthagiri, @robrap thanks for this note. I jumped to changing this repo after direction from Dave and did not know that the RequestMetricsMiddleware repo existed. I'm sure that a refactor is possible, but I'd prefer to check in with @mulby and team and then figure out how to proceed from there.

Reply:
To be clear - a refactor and follow-on can happen after the weekend.  Let's take this as an "IOU" to clean it up - as we don't want to necessarily block the team's immediate deadline.  Hopefully, it's just a matter of moving this metric to the other middleware - and not much more.